src/newgrf.cpp

```diff
@@ -2676,24 +2690,33 @@ static ChangeInfoResult TownHouseChangeInfo(uint hid, int numinfo, int prop, Byt
  * @param gvid ID of the global variable. This is basically only checked for zerones.
  * @param numinfo Number of subsequent IDs to change the property for.
  * @param buf The property value.
- * @param[in,out] translation_table Storage location for the translation table.
+ * @param gettable Function to get storage for the translation table.
  * @param name Name of the table for debug output.
  * @return ChangeInfoResult.
  */
 template <typename T>
-static ChangeInfoResult LoadTranslationTable(uint gvid, int numinfo, ByteReader &buf, std::vector<T> &translation_table, const char *name)
+static ChangeInfoResult LoadTranslationTable(uint gvid, int numinfo, ByteReader &buf, std::function<std::vector<T> &(GRFFile &)> gettable, std::string_view name)
```

Comment:
historically we've avoided std::function because of various overheads. Guess template parameters are out of the question with lambdas? Maybe passing some sort of enum value as a template param and then filtering with an if constexpr or something?

Reply:
Probably the overhead doesn't matter here given it's only used on load and only for the lifetime for the call. However given it's already a templated function, I simply turned the function into a templated argument instead.